### DISCUSSION TITLE
How to create a minimal tenant without admin UI?

### DISCUSSION THREAD

**Discussion by hieucd04** [OrchardCMS/OrchardCore#11549] [Q&A] [answered]
I'm using `Orchard Core CMS` with the `default` tenant up and running. The `default` tenant has `/admin`.
Now, I want to create a new tenant to host a static website (only `HTML, CSS & JS` files).
Here are my requirements:
    - New tenant must not have the admin UI (`/admin`)
    - New tenant must be as minimal as possible. Preferably, all modules are turn off by default.
    - I can selectively enable modules which I need (ex: the ones that I wrote myself).
    - If possible, I want to create the new tenant from code (ex: appsettings.json)

Is there anyway to do this?

**Comment by ns8482e** [comment]
You can create a setup recipe that  can include only features that you want enabled for tenants and create tenant using that recipe.

Using auto setup module you can setup tenant using appsettings.json

**Comment by hishamco** [comment, marked as answer]
Please check [this recipe](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/dev/src/OrchardCoreContrib.Modules.Web/Recipes/OCC.recipe.json) and [appsettings.json](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/dev/src/OrchardCoreContrib.Modules.Web/appsettings.json) that I created for OCC modules

**Comment by hieucd04** [comment]
Comments from hishamco and ns8482e gave me a really good starting point.
For anyone who needs it, here is my recipe with minimal number of modules enabled:
```
{
  "name": "Minimal",
  "displayName": "Minimal",
  "description": "",
  "author": "hieucd04",
  "version": "1.0.0",
  "issetuprecipe": true,
  "categories": [ "minimal" ],
  "tags": [ "minimal" ],
  "steps": [
    {
      "name": "feature",
      "disable": [],
      "enable": [
        "OrchardCore.Settings", // Required for dependency injection to work
        "OrchardCore.HomeRoute", // Provide a HomeController. If you don't need HomeController you can remove this
        "Orchard.Storybook" // I wrote this module to provide functionalities specifically for my site
      ]
    }
  ]
}

```

This recipe can then be used when creating new tenants from Admin UI or [AutoSetup](https://docs.orchardcore.net/en/dev/docs/reference/modules/AutoSetup/).

**Reply by hishamco** [reply]
The above answered your, what you did is minimize the enabled features ;)
